internal/provider/resource_auth0_custom_domain.go

```diff
@@ -74,14 +74,34 @@ func newCustomDomain() *schema.Resource {
 					},
 				},
 			},
+			"custom_client_ip_header": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"cf-connecting-ip", "x-forwarded-for", "true-client-ip", "",
+				}, false),
+				Description: "The HTTP header to fetch the client's IP address. " +
+					"Cannot be set on auth0_managed domains.",
+			},
+			"tls_policy": {
```

Comment:
We're adding this field as well for completeness as the PATCH endpoint accepts it as well.

Reply:
I was planning to add this field in my next PR but good to have them both here!